examples/gno.land/p/demo/timelock/timelock.gno

```diff
@@ -0,0 +1,237 @@
+// Package timelock provides a library for scheduling, cancelling, and
+// executing time-locked operations in Gno. It ensures that
+// operations are only carried out after a specified delay and offers
+// mechanisms for managing and verifying the status of these operations.
+// This package leverages an AVL tree for efficient management of timestamps
+// and integrates role-based access control for administrative tasks.
+//
+// Example Usage:
+//
+//  import "gno.land/p/demo/timelock"
+//  import "gno.land/p/demo/accesscontrol"
+//
+//  Initialize timelock utility with an AVL tree and access control.
+//  timestamps := avl.NewTree()
+//  adminRole := accesscontrol.NewRole("admin", std.Address("admin-address"))
+//  timeLockUtil := timelock.NewTimeLockUtil(timestamps, adminRole, 30)
+//
+//  Schedule an operation with a delay of 60 seconds.
+//  id := seqid.ID()
+//  timeLockUtil.Schedule(id, 60)
+//
+//  Check if an operation is pending.
+//  isPending := timeLockUtil.IsPending(id)
+//
+//  Execute the operation when it is pending.
+//  if timeLockUtil.IsPending(id) {
+//      timeLockUtil.Execute(id)
+//  }
+//
+//  Update the minimum delay for future operations.
+//  timeLockUtil.UpdateDelay(45)
+
+package timelock
+
+import (
+	"std"
+	"strconv"
+	"time"
+
+	"gno.land/p/demo/accesscontrol"
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+)
+
+// Represents the status of a planned operation
+type OperationState int
+
+const (
+	Unset OperationState = iota
+	Pending
+	Ready
+	Done
+)
+
+func (os OperationState) StateToString() string {
+	switch os {
+	case Unset:
+		return "Unset"
+	case Pending:
+		return "Pending"
+	case Ready:
+		return "Ready"
+	case Done:
+		return "Done"
+	default:
+		return "Unknown"
+	}
+}
+
+// OperationStatus represents the status of an operation
+type OperationStatus struct {
+	sheduleTime int64
+	isDone      bool
+}
+
+// TimeLockUtil stores the necessary parameters for the timelock operations
+type TimeLockUtil struct {
```

Comment:
Why Util?

Reply:
`Util` to represent a "toolbox" of essential parameters and methods needed for managing locking operations
You don't find the word explicit enough ? @thehowl

Reply:
No, it just seems redundant. Also, these "toolboxes" are generally discouraged in Go (and Gno): https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

Reply:
Thank for shared the article on `avoid-package-names`. 
As a result changed `TimelockUtil` to `Timelock`